### PR TITLE
add tests for AuthenticateDatabase data source

### PR DESCRIPTION
**Pull request by csaltmountain**

### CODE REVIEW

File: app/data-sources/authenticate/AuthenticateDatabase.js [hunk 1]
```diff
@@ -1,49 +1,48 @@
...
-
-const readUsername = process.env.AUTHENTICATE_DB_USERNAME
-const readPassword = process.env.AUTHENTICATE_DB_PASSWORD
-const sequelizeRead = new Sequelize(databaseName, readUsername, readPassword, {
-  host: serverHost,
+const dbOptions = {
+  database: process.env.AUTHENTICATE_DB_SCHEMA,
+  host: process.env.AUTHENTICATE_DB_HOST,
   dialect: 'mssql',
   dialectOptions: { options: { encrypt: false } },
```
Comment (rkdefra) [inline app/data-sources/authenticate/AuthenticateDatabase.js:8]: `encrypt: false` really!?

File: test/data-sources/autenticate.test.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  })
+
+  test('should create audit record when getting answers', async () => {
+    const queryMock = jest.fn()
+    Sequelize.mockImplementation(() => ({
+      define: () => ({ findOne: jest.fn() }),
+      query: queryMock
+    }))
+
+    jest.useFakeTimers().setSystemTime(new Date('2024-08-28T00:00:00.000Z'))
```
Comment (rkdefra) [inline test/data-sources/autenticate.test.js:74]: Do you need to reset this in an `after` block?

**Review by rkdefra — APPROVED** [review]
Just a couple of minor queries, but looks good :+1:

File: app/data-sources/authenticate/AuthenticateDatabase.js (hunk 1 above)
Comment (csaltmountain) [reply app/data-sources/authenticate/AuthenticateDatabase.js:8]: Not sure… @defra-dom do you know? I just copied over the settings from your original implementation. When working locally with docker, doesn't seem to make much difference if `true` or `false`.

Comment (defra-dom) [reply app/data-sources/authenticate/AuthenticateDatabase.js:8]: cant remember, will be somthing to do with connecting to auth db in live

**Review by defra-dom — DISMISSED** [review]
LGTM

**Review by defra-dom — APPROVED** [review]

**Pull request merged**